### PR TITLE
Add inspiredminds/contao-news-twitter

### PR DESCRIPTION
**Pull request by fritzmg**
This adds documentation for the proprietary inspiredminds/contao-news-twitter extension.

### CODE REVIEW

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
+---
+title: Contao News Twitter
+menuTitle: News Twitter
+description: Contao News Twitter ist eine kostenpflichtige Erweiterungen zur Synchronisierung von Tweets mit einem Nachrichtenarchiv.
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
description: Contao News Twitter ist eine kostenpflichtige Erweiterung zur Synchronisierung von Tweets mit einem Nachrichtenarchiv.
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+{{% /expand %}}
+
+Nachdem diese Änderungen gemacht wurden, kann nun ein `composer update` auf der 
+Kommandozeile oder eine Paketaktualisierung im Contao Manager durchgeführt werden. 
+Danach wie gewohnt das Contao Install Tool aufrufen, um die Datenbank zu aktualisieren.
+
+
+## Konfiguration
+
+Um diese Erweiterung zu konfigurieren muss zuerst eine sogenannte Twitter App« erzeugt
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
Um diese Erweiterung zu konfigurieren, muss zuerst eine sogenannte »Twitter App« erzeugt 
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+Um diese Erweiterung zu konfigurieren muss zuerst eine sogenannte Twitter App« erzeugt 
+werden. Die Details dieser App werden dann für die Konfiguration im Backend benötigt.
+
+
+### Twitter App anlegen
+
+1. Gehe zu [apps.twitter.com](https://apps.twitter.com/).
+2. Klicke auf _Create New App_.
+3. Fülle die Basisinformationen aus.
+4. Befüle die _Callback URL_ mit der Contao Backend URL, z. B. `http://www.example.org/contao`.
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
4. Befülle die _Callback URL_ mit der Contao Backend URL, z. B. `https://www.example.org/contao`.
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+Öffne die Einstellungen deines Nachrichtenarchivs. In der Sektion _Twitter sync_ 
+kann __Twitter sync__ aktiviert werden. Nun stehen folgende Optionen zur Verfügung:
+
+__Authentifizierung__: Bei Klick auf das Profilbild bzw. Fragezeichen kann man sich
+bei Twitter (re-)authentifizieren. Dies ist notwendig, wenn man Nachrichten automatisch
+zu Twitter posten lassen möchte. Es ist _nicht_ notwendig sich zu authentifizieren,
+wenn man nur Tweets von einem Profil oder einem Hashtag holen möchte.
+
+__Profil__: Das Twitter Profil, von diesem Tweets geholt oder Tweets gepostet werden
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
__Profil__: Das Twitter Profil, von dem Tweets geholt oder Tweets gepostet werden 
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+__Hasthtags__: Liste an Hashtags durch Leerzeichen oder Komma getrennt, welche von
+Twitter geholt werden sollen. _Achtung:_ wenn auch ein Twitter Profil definiert 
+wurde, dann agiert diese Liste als Filter für die Tweets des Profiles. Ohne Angabe
+eines Profils werden alle Tweets zu den angegebenen Hashtags geholt.
+
+__Tweets holen__: Nur wenn diese Option aktiv ist, werden Tweets vom angegebenen
+Profil bzw. von den angegebenen Hashtags geholt.
+
+__Tweets veröffentlichen__: Nur wenn dies Option aktiv ist, werden Nachtichten auch
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
__Tweets veröffentlichen__: Nur wenn dies Option aktiv ist, werden Nachrichten auch 
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+__Tweets veröffentlichen__: Nur wenn dies Option aktiv ist, werden Nachtichten auch 
+automatisch veröffentlicht, wenn sie von Twitter geholt wurden.
+
+
+## Benutzung
+
+### Tweets holen
+
+Sobald ein Nachrichtenarchiv mit der Option __Tweets holen__ konfiguriert ist, wird
+über den Contao Cronjob stündlich auf neue Tweets überprüf.
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
über den Contao Cronjob stündlich auf neue Tweets überprüft.
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+In der Übersicht der Nachrichtenarchive gibt es einen Link, mit dem man die Synchronisation
+der Nachrichtenarchive manuell anstoßen kann.
+
+
+## Hooks
+
+### `processTweet`
+
+Die Erweiterung prozessiert Tweets und versucht den Tweet in eine passendes Format
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
Die Erweiterung prozessiert Tweets und versucht den Tweet in ein passendes Format
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+für eine Contao Nachricht zu verwandeln. Falls die finalen Daten für die Nachricht
+angepasst werden sollen, kann der `processTweet` Hook benutzt werden. Als Rückgabewert
+wird ein Array erwartet, dass die finalen Daten enthält. Soll ein Tweet nicht als
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
wird ein Array erwartet, das die finalen Daten enthält. Soll ein Tweet nicht als
```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+1. _array_ `$arrData` Die bereits prozessierten Daten für die Nachricht.
+2. _object_ `$objTweet` Die originalen Daten des Tweets.
+3. _object_ `$objArchive` Das Nachrichtenarchiv Objekt.
+
+
+### `changeTwitterMessage`
+
+Wenn eine Contao Nachricht als Tweet gepostet werden soll, wird entweder die Überschrift
+oder der angegebene Text verwenden. Über den `changeTwitterMessage` Hook kann diese
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md]:
```suggestion
oder der angegebene Text verwendet. Über den `changeTwitterMessage` Hook kann diese
```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+To configure this extension, a "Twitter App" needs to created first. The details 
+of this app are then needed for the configuration in the back end.
+
+
+### Creating a Twitter App
+
+1. First you need to go to [apps.twitter.com](https://apps.twitter.com/).
+2. Click on _Create New App_.
+3. Fill out the basic information.
+4. Fill out the _Callback URL_ with the Contao back end URL, e.g. `http://www.example.org/contao`.
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md]:
```suggestion
4. Fill out the _Callback URL_ with the Contao back end URL, e.g. `https://www.example.org/contao`.
```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+```
+{{% /expand %}}
+
+After making this adjustment, run a `composer update` on the command line or update 
+your packages via the Contao Manager. Then open the Contao Install Tool to update the database as usual.
+
+
+## Configuration
+
+To configure this extension, a "Twitter App" needs to created first. The details
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md:106]:
```suggestion
To configure this extension, a »Twitter App« needs to created first. The details 
```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+1. First you need to go to [apps.twitter.com](https://apps.twitter.com/).
+2. Click on _Create New App_.
+3. Fill out the basic information.
+4. Fill out the _Callback URL_ with the Contao back end URL, e.g. `http://www.example.org/contao`.
+5. Click on _Create your Twitter application_.
+
+
+### Configure the Consumer Key and Consumer Secret in Contao
+
+In Contao, go to _System_ » _Settings_, scroll down to the the _Twitter App_ section,
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md]:
```suggestion
In Contao, go to _System_ » _Settings_, scroll down to the _Twitter App_ section, 
```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+__Authentication__: by clicking on the profile picture (or question mark), you can 
+(re-)authenticate with your Twitter account. This is necessary if you want to tweet 
+your news entries to your profile automatically. It is _not_ necessary to authenticate, 
+if you only want to fetch tweets from hashtags or Twitter profiles.
+
+__Profile__: the Twitter profile where tweets should either be fetched from or posted 
+to.
+
+__Hasthtags__: space or comma separated list of hashtags that should be fetched 
+from Twitter. _Note:_ if you also defined a _Profile_ this lists acts as a filter
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md]:
```suggestion
from Twitter. _Note:_ if you have also defined a _Profile_ this lists acts as a filter 
```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+be set to _published_.
+
+
+## Usage
+
+### Fetch Tweets
+
+Once you have configured the news archive and also enabled the __Fetch tweets__ 
+option the extension will check for new Tweets hourly via Contao's cronjob.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md:157]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+- the news entry must be published
+
+You can also define a custom message for the tweet. If no such message is defined, 
+the news entry's headline is used by default.
+
+If the news entry has a teaser image, that image will also be attached to the tweet.
+
+The extension will check for new news entries to be published minutely via Contao's 
+cronjob.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md:174]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+## Hooks
+
+### `processTweet`
+
+The extension processes a tweet and tries to convert it to a fitting Contao news 
+entry as best as it can. If you want to customize the final data for a news entry 
+of a tweet, you can use the `processTweet` hook. It expects an array containing 
+the final news entry data as the return value. If the return value equals to false, 
+no news entry will be created.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md:190]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+#### Parameters
+
+1. _array_ `$arrData` The already processed data which will be used for the new 
+   news entry.
+2. _object_ `$objTweet` The original tweet data.
+3. _object_ `$objArchive` The news archive object.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md:198]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.en.md [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+### `changeTwitterMessage`
+
+When posting a Contao news entry as a tweet, the extension either uses the teaser 
+or the specified message of the news entry. If you want to automatically provide 
+a different message, you can use the `changeTwitterMessage` hook. It expects the final message as the return value.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.en.md:205]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 19]
```diff
@@ -0,0 +1,221 @@
...
+## Hooks
+
+### `processTweet`
+
+Die Erweiterung prozessiert Tweets und versucht den Tweet in eine passendes Format
+für eine Contao Nachricht zu verwandeln. Falls die finalen Daten für die Nachricht
+angepasst werden sollen, kann der `processTweet` Hook benutzt werden. Als Rückgabewert
+wird ein Array erwartet, dass die finalen Daten enthält. Soll ein Tweet nicht als
+Nachricht gespeichert werden, kann etwas leeres zurückgegeben werden.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md:194]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.de.md [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+automatisch veröffentlicht, wenn sie von Twitter geholt wurden.
+
+
+## Benutzung
+
+### Tweets holen
+
+Sobald ein Nachrichtenarchiv mit der Option __Tweets holen__ konfiguriert ist, wird
+über den Contao Cronjob stündlich auf neue Tweets überprüf.
+
```
Comment (xchs) [inline docs/manual/extensions/news-twitter-sync.de.md:163]:
```suggestion


```

File: docs/manual/extensions/news-twitter-sync.en.md (hunk 11 above)
Comment (fritzmg) [reply docs/manual/extensions/news-twitter-sync.en.md:106]: English uses `"`.

**Comment by fritzmg** [comment]
@xchs thank you so much for reviewing, as always :)

**Pull request merged**